File: SDKs/Aspose.Cells_Cloud_SDK_For_Ruby/lib/aspose_cells_cloud/models/date_time_group_item.rb
module AsposeCellsCloud
  # 
  class DateTimeGroupItem < BaseObject
    attr_accessor :date_time_grouping_type, :day, :hour, :minute, :month, :second, :year
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'date_time_grouping_type' => :'DateTimeGroupingType',
        
        # 
        :'day' => :'Day',
        
        # 
        :'hour' => :'Hour',
        
        # 
        :'minute' => :'Minute',
        
        # 
        :'month' => :'Month',
        
        # 
        :'second' => :'Second',
        
        # 
        :'year' => :'Year'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'date_time_grouping_type' => :'String',
        :'day' => :'Integer',
        :'hour' => :'Integer',
        :'minute' => :'Integer',
        :'month' => :'Integer',
        :'second' => :'Integer',
        :'year' => :'Integer'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'DateTimeGroupingType']
        self.date_time_grouping_type = attributes[:'DateTimeGroupingType']
      end
      
      if attributes[:'Day']
        self.day = attributes[:'Day']
      end
      
      if attributes[:'Hour']
        self.hour = attributes[:'Hour']
      end
      
      if attributes[:'Minute']
        self.minute = attributes[:'Minute']
      end
      
      if attributes[:'Month']
        self.month = attributes[:'Month']
      end
      
      if attributes[:'Second']
        self.second = attributes[:'Second']
      end
      
      if attributes[:'Year']
        self.year = attributes[:'Year']
      end
      
    end

  end
end

File: SDKs/Aspose.Cells-Cloud-SDK-for-.NET/src/Com/Aspose/cells/Model/DateTimeGroupItem.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Com.Aspose.Cells.Model {
  public class DateTimeGroupItem {
    public string DateTimeGroupingType { get; set; }

    public int? Day { get; set; }

    public int? Hour { get; set; }

    public int? Minute { get; set; }

    public int? Month { get; set; }

    public int? Second { get; set; }

    public int? Year { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class DateTimeGroupItem {\n");
      sb.Append("  DateTimeGroupingType: ").Append(DateTimeGroupingType).Append("\n");
      sb.Append("  Day: ").Append(Day).Append("\n");
      sb.Append("  Hour: ").Append(Hour).Append("\n");
      sb.Append("  Minute: ").Append(Minute).Append("\n");
      sb.Append("  Month: ").Append(Month).Append("\n");
      sb.Append("  Second: ").Append(Second).Append("\n");
      sb.Append("  Year: ").Append(Year).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }
